model/annotations/task_annotations.go

```diff
@@ -211,15 +211,23 @@ func UpdateAnnotation(a *TaskAnnotation, userDisplayName string) error {
 func PatchAnnotation(a *TaskAnnotation, userDisplayName string, upsert bool) error {
 	existingAnnotation, err := FindOneByTaskIdAndExecution(a.TaskId, a.TaskExecution)
 	if err != nil {
-		return errors.Wrapf(err, "finding task annotation for task id '%s'", a.TaskId)
+		return errors.Wrapf(err, "finding task annotation")
 	}
-	if existingAnnotation == nil {
+	if existingAnnotation == nil && upsert {
 		if !upsert {
```

Comment:
at this point we know upsert is true because of the if on 216

Reply:
ah it wasnt supposed to check it in the first if